### PR TITLE
Classlist

### PR DESCRIPTION
**Pull request by mafewer**
## Problem Statement
This is the same PR as the draft from last week, just with the updated variables and such for Moaaz's table. Sorry about the wait.

## Solution
Given a course code, in this case 6400, you receive a JSON response which has the success, error, illegal field as usual, but also a list of the data associated with each student, that being their profilePictures, usernames, etc

## Testing
https://drive.google.com/file/d/1Mf8bGDslAvPUPuN3HYdw6M-woS4-dLEd/view?usp=drive_link

### CODE REVIEW

File: services/get-students.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// File: /services/get_students.php
+
+// Set the content type to JSON
+header('Content-Type: application/json');
+
+// Shared secret for authorization
+$sharedSecret = "gradeplus";
+
+// Check authorization
+if ($authorize !== $sharedSecret) {
```
Comment (akashcraft) [inline services/get-students.php]: I do not see any inputs being received from front-end. This variable would be undefined. `$authorize` should be `$_POST["authorize"]`

File: services/get-students.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+// Database credentials
+$db_host = 'localhost';
+$db_user = 'gradeplusclient';
+$db_pass = 'gradeplussql';
+$db_name = 'gradeplus';
+
+// Create a new MySQLi connection
+$conn = new mysqli($db_host, $db_user, $db_pass, $db_name);
```
Comment (akashcraft) [inline services/get-students.php]: Remove single use variables and directly put in values in `$conn`
```suggestion
$conn = new mysqli('localhost','gradeplusclient' , 'gradeplussql', 'gradeplus');
```

File: services/get-students.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    enrollment.courseCode = ?
+";
+
+
+// Prepare the statement
+$stmt = $conn->prepare($sql);
+if (!$stmt) {
+    sendResponse(0, 1, [], "Preparation failed: " . $conn->error);
+}
+$stmt->bind_param("s", $course_code);
```
Comment (akashcraft) [inline services/get-students.php]: Possible NPE on `$course_code`

**Review by akashcraft — CHANGES_REQUESTED** [review]

**Comment by mafewer** [comment]
Updated video test too https://drive.google.com/file/d/1gQWg4WODnx5KQmkTjfBDxvGpHx19-xv4/view?usp=sharing

**Review by akashcraft — APPROVED** [review]
Excellent!

**Review by MoaazElshabasy — COMMENTED** [review]
Neat

**Review by MoaazElshabasy — APPROVED** [review]
Neat

**Pull request merged**